src/hashtable.c

```diff
@@ -719,6 +722,38 @@ static int findKeyInBucketSSE2(hashtable *ht, bucket *b, uint8_t h2, const void
 }
 #endif
 
+#if HAVE_ARM_NEON && ENTRIES_PER_BUCKET <= 8
+/* 32 bit architectures would require a different implementation
+ * consider that even if they had Neon SIMD support,
+ * they have 12 entries per bucket and only 32-bit scalar registers */
+
+static inline __attribute__((hot, always_inline)) int popMatchBitmask(uint64_t *mask) {
+    /* one byte (8 bits) per item - either 0x80 or 0x00 */
+    int pos = __builtin_ctzll(*mask) >> 3;
+    *mask &= (*mask - 1); /* clear lowest set bit */
+    return pos;
+}
+
+static __attribute__((hot)) int findKeyInBucketNeon(hashtable *ht, bucket *b, uint8_t h2, const void *key, int table, int *pos_in_bucket, int *table_index) {
```

Comment:
Small comment about "hot" attribute: although we do not provide explicit flag for PGO it can still be used while compiling valkey and define this can have impact on PGO as it will override the profile collected data. I am not sure how much gain we got from using this attribute? did you observe some difference when using it?

Reply:
I was playing around with compiling with this on my M4 macbook and it didn't compile any differently. It just inlined it. It inlined it without hot or without the "always_inline" that was also placed on the other function as well. Our general guidance has always been don't add attributes unless they demonstrate improvement.